bentoml/_internal/configuration/containers.py

```diff
@@ -223,24 +224,21 @@ def tracer(
 
             return NoopTracer()
 
-    logging_file_directory = t.cast(
-        Provider[t.Callable[[], str]],
+    logging_file_directory = providers.Factory[str](
+        lambda default, customized: customized or default,
         providers.Factory(
-            lambda default, customized: customized or default,
-            providers.Factory(
-                os.path.join,
-                bentoml_home,
-                "logs",
-            ),
-            config.logging.file.directory,
+            os.path.join,
+            bentoml_home,
+            "logs",
         ),
+        config.logging.file.directory,
     )
 
 
 BentoMLContainer = BentoMLContainerClass()
 
 
-@container
+@attr.s
```

Comment:
Same as above

Reply:
also doesn't seem like pyright is complaining